frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -662,6 +662,45 @@ final class BloopBspServices(
     }
   }
 
+  def jvmTestEnvironment(params: bsp.JvmEnvironmentParams): BspEndpointResponse[bsp.JvmEnvironmentResult] = {
+    ifInitialized(None) { (state: State, logger: BspServerLogger) =>
+
+      mapToProjects(params.targets, state) match {
+        case Left(error) =>
+          logger.error(error)
+          Task.now((state, Left(JsonRpcResponse.invalidRequest(error))))
+
+        case Right(projects) =>
+          val environmentEntries: List[Either[String, JvmEnvironmentEntry]] = for {
+            (id, project) <- projects.toList
+            dag = state.build.getDagFor(project)
+            fullClasspath = project.fullClasspath(dag, state.client).map(_.toString)
+            environmentVariables = state.commonOptions.env.toMap
+            workingDirectory = state.commonOptions.workingDirectory
+          } yield {
+            project.platform match {
+              case Platform.Jvm(config, _, _) => Right(
+                bsp.JvmEnvironmentEntry(id, fullClasspath.toList,
+                  config.javaOptions.toList, workingDirectory, environmentVariables)
+              )
+              case _ => Left(s"Not a JVM project: ${project.name}")
+            }
+          }
+
+          def sequenceListOfEithers[L, R](agg: List[R], col: List[Either[L, R]]): Either[L, List[R]] = {
```

Comment:
Maybe I should change it to `Traversable[_].sequence` from cats. But it would require `-Ypartial-unification`. Please let me know which approach you prefer.

Reply:
I think cats is on the classpath because of the circe dependency via lsp4s. I think it's best not to use cats APIs directly, however.

Reply:
Not applicable if we silently ignore non-jvm configs